api/handlers.go

```diff
@@ -330,6 +330,40 @@ func getVolume(c *context, w http.ResponseWriter, r *http.Request) {
 	httpError(w, fmt.Sprintf("No such volume: %s", name), http.StatusNotFound)
 }
 
+func getUsedVolumes(containers []*cluster.Container) map[string]bool {
+	usedVolumes := make(map[string]bool)
+	for _, container := range containers {
+		for _, mount := range container.Mounts {
+			if mount.Type == "volume" {
+				usedVolumes[container.Engine.Name+"/"+mount.Name] = true
+			}
+		}
+	}
+	return usedVolumes
+}
+
+func getFilteredVolumes(containers []*cluster.Container, volumes []*apitypes.Volume, dangling bool) []*apitypes.Volume {
+	usedVolumes := getUsedVolumes(containers)
+	var filteredVolumes []*apitypes.Volume
+	// volume name allowed char [a-zA-Z0-9][a-zA-Z0-9_.-]
+	// splitting on / is safe to keep the original output intact
+	stripEngine := func(vol *apitypes.Volume) *apitypes.Volume {
+		//don't change local volume names "engine_name/volume_name"
```

Comment:
I meant that you should add a comment explaining why you're doing this.

Reply:
updated